src/OrleansAzureUtils/Storage/AzureTableDataManager.cs

```diff
@@ -467,57 +437,93 @@ public AzureTableDataManager(string tableName, string storageConnectionString, L
-                        while (querySegment == null || querySegment.ContinuationToken != null)
-                        {
-                            querySegment = await cloudTableQuery.ExecuteSegmentedAsync(querySegment?.ContinuationToken);
-                            list.AddRange(querySegment);
-                        }
-
-                        return list;
-                    };
-
-                    IBackoffProvider backoff = new FixedBackoff(AzureTableDefaultPolicies.PauseBetweenTableOperationRetries);
-
-                    List<T> results = await AsyncExecutorWithRetries.ExecuteWithRetries(
-                        counter => executeQueryHandleContinuations(),
-                        AzureTableDefaultPolicies.MaxTableOperationRetries,
-                        (exc, counter) => AzureStorageUtils.AnalyzeReadException(exc.GetBaseException(), counter, TableName, Logger),
-                        AzureTableDefaultPolicies.TableOperationTimeout,
-                        backoff);
-
-                    // Data was read successfully if we got to here                    
-                    return results.Select(i => Tuple.Create(i, i.ETag)).ToList();
-                }
-                catch (Exception exc)
+                return await ReadTableEntriesAndEtagsWithTableQueryAsync(cloudTableQuery);
+            }
+            finally
+            {
+                CheckAlertSlowAccess(startTime, operation);
+            }
+        }
+#endif
+
+        /// <summary>
+        /// Read data entries and their corresponding eTags from the Azure table.
+        /// </summary>
+        /// <param name="filter">filter string to use for querying the table and filtering the results.</param>
+        /// <returns>Enumeration of entries in the table which match the query condition.</returns>
+        public async Task<IEnumerable<Tuple<T, string>>> ReadTableEntriesAndEtagsWithFilterStringAsync(string filter)
+        {
+            const string operation = "ReadTableEntriesAndEtags";
+            var startTime = DateTime.UtcNow;
+            try
+            {
+                TableQuery<T> cloudTableQuery = filter == null
+                    ? new TableQuery<T>()
+                    : new TableQuery<T>().Where(filter);
+                return await ReadTableEntriesAndEtagsWithTableQueryAsync(cloudTableQuery);
+            }
+            finally
+            {
+                CheckAlertSlowAccess(startTime, operation);
+            }
+        }
+        /// <summary>
+        /// Read data entries and their corresponding eTags from the Azure table.
+        /// </summary>
+        /// <param name="filter">Predicate function to use for querying the table and filtering the results.</param>
+        /// <returns>Enumeration of entries in the table which match the query condition.</returns>
+        private async Task<IEnumerable<Tuple<T, string>>> ReadTableEntriesAndEtagsWithTableQueryAsync(TableQuery<T> cloudTableQuery)
```

Comment:
If we're always returning a list, and especially given this is a private method, we should just return the concrete type: `List<T>` would be preferable. I see that the usages of this method convert it to a list anyway.

I'd understand if you consider this out of scope for this PR, since you're really just converting an existing method which follows this pattern.